sdk/spring-sdk/src/it/java/com/example/wiring/actions/echo/EchoAction.java

```diff
@@ -37,4 +40,16 @@ public Effect<Message> stringMessage(@PathVariable String msg) {
     String response = this.parrot.repeat(msg);
     return effects().reply(new Message(response));
   }
+
+  @GetMapping("/echo/repeat/{msg}/times/{times}")
+  public Flux<Effect<Message>> stringMessage(
+      @PathVariable String msg, @PathVariable Integer times) {
+
+    Stream<Effect<Message>> messages =
+        IntStream.rangeClosed(1, times)
+            .mapToObj(i -> new Message(parrot.repeat(msg)))
+            .map(m -> effects().reply(m));
+
+    return Flux.fromStream(messages);
```

Comment:
What do you mean by a stream processing that is actually async?

Reply:
Do you mean to consume the stream element by element instead of dumping it to a collection? 

I can pick one element at a time, but I won't be able to say if the elements have been eagerly fetched or if I'm picking then on demand. 

I could have used an infinite stream and interrupt the consumption. Somehow, if we were dumping it in memory we would get an OOM (or not). So didn't look like an effective test. 

Anyway, please clarify what you have in mind about it.

Reply:
I mean that this Java stream isn't actually async but more like an iterator that can be consumed singlethreaded, so testing something closer to Akka source.unfold from the reactor APIs that actually does some async thread jumps for good measure.

Reply:
Maybe an intermediate Flux.flatMap would be good enough.

Reply:
Oh, I see what you mean. We are in the `EchoAction`. I thought this was the test side where I make a collection out of it.